### DISCUSSION TITLE
milvus supports setting a primary key ID when inserting records and enables the functionality to update the record if the ID already exists or to create a new record if the ID does not exist?

### DISCUSSION THREAD

**Discussion by qiankunli** [milvus-io/milvus#35069] [Q&A and General discussion]
In our process of indexing documents, we have an automatic retry mechanism in place. If a document fails to process halfway and we need to re-enter it into the database, without support for the aforementioned 'update if exists, create if not' functionality, there is a risk of duplicating chunks.

**Comment by yhmo** [comment]
There are two methods to insert records:
1. insert()
This method doesn't check duplication, it directly inserts the record with a specified id, even the id is already duplicated in the collection.

2. upsert()
If the id is already exist, this method will perform a delete action to the existing record, and insert a new record to replace the old one.

upsert() is much heavier than insert(). Ideally, it is better that each record has a unique id and is inserted by insert() method.
